### ISSUE TITLE
Error in Cross entropy

### ISSUE THREAD

**Issue by jianlong-yuan** [open-mmlab/mmsegmentation#886]
If there is an ignore label, the results of these two calculations are not the same, and the current implementation will average the ignore area as well.

logits = torch.rand(10, 19, 20, 20)
y = torch.randint(0, 19, (10, 20, 20))
mask = torch.randint(0, 2, (10, 20, 20))
y[mask==1] = 255
loss= F.cross_entropy(logits, y, ignore_index=255, reduction='mean')
print(loss)
loss= F.cross_entropy(logits, y, ignore_index=255, reduction='none').mean()
print(loss)

**Comment by MengzhangLI** [comment]
Hi, sorry for late reply.

Related PR is here: https://github.com/open-mmlab/mmsegmentation/pull/578

We would consider to merge it as soon as possible.

**Comment by MeowZheng** [comment]
solved in https://github.com/open-mmlab/mmsegmentation/pull/1409
